src/server.ts

```diff
@@ -7,21 +7,33 @@ import cors from "cors";
 import schema from "./schema";
 import "reflect-metadata";
 import { Database } from "./config/database";
+import dotenv from "dotenv";
 
-Database.config()
+const Server = async () => {
+  dotenv.config({ path: process.env.TEST_RUNNING === "TRUE" ? "./.test.env" : "./.env" });
 
-const app = express();
-const server = new ApolloServer({
-  schema,
-  validationRules: [depthLimit(7)],
-});
+  const DatabaseProps = {
+    port: Number(process.env.DATABASE_PORT),
+    username: String(process.env.DATABASE_USERNAME),
+    password: String(process.env.DATABASE_PASSWORD),
+    databaseName: String(process.env.DATABASE_NAME),
+  };
 
-app.use(cors());
-app.use(compression());
-server.applyMiddleware({ app, path: "/graphql" });
+  await Database.config(DatabaseProps).then((): void => console.log(process.env.DATABASE_CONNECTED_MESSAGE));
 
-const httpServer = createServer(app);
+  const app = express();
+  const server = new ApolloServer({
+    schema,
+    validationRules: [depthLimit(7)],
+  });
 
-httpServer.listen({ port: 4000 }, (): void =>
-  console.log("Backend onboarding is now runnind on http://localhost:4000/graphql")
-);
+  app.use(cors());
+  app.use(compression());
+  server.applyMiddleware({ app, path: "/graphql" });
+
+  const httpServer = createServer(app);
+
+  httpServer.listen({ port: 4000 }, (): void => console.log(process.env.SERVER_CONNECTED_MESSAGE));
```

Comment:
A porta é uma variável de ambiente também. Geralmente quando fazemos deploy em alguma plataforma, pode ser que ela mesma escolha uma porta e coloque na variável `PORT`. Uma coisa q vc pode fazer é ter como padrão a 4000:

 ```suggestion
  httpServer.listen({ port: process.env.PORT ?? 4000 }, (): void => console.log(process.env.SERVER_CONNECTED_MESSAGE));
```

Reply:
Com isso, talvez as envs `BASE_URL` e `SERVER_CONNECTED_MESSAGE` pudessem nem ser variáveis de ambiente, mas entendo que vc quis fazer isso pra diferenciar entre ambiente local e test, então sem problemas em manter tbm.

Reply:
Legal! Deletei a BASE_URL e deixei com a PORT. Mantive a mensagem do server mais por diferenciar mesmo e garantir que estou no ambiente certo